metacat-common-server/src/main/java/com/netflix/metacat/common/server/properties/Config.java

```diff
@@ -266,4 +266,10 @@
      * @return The partition topic ARN or null if no property set
      */
     String getSnsTopicPartitionArn();
+
+    /**
+     * Use default user metadata service.
+     * @return The flag to turn on mysql based user metadata service.
+     */
+    boolean userMetadataServiceEnabled();
```

Comment:
The adapter abstraction mentioned above will work. I think the second option, as standard constructor injection, is more proper here. Say we have two beans defined as the implementation of userMetadataService. The default one is the no-op service, which we mark as primary (matchIfMissing). The mysql one can be injected if we set the property explicitly. The property on the profile/property is the name of the implementation, e.g., 'mysql', not 'enabled'.